misc/python/materialize/cli/ci_logged_errors_detect.py

```diff
@@ -55,13 +54,18 @@
     | clusterd:\ fatal: # startup failure
     | error:\ Found\ argument\ '.*'\ which\ wasn't\ expected,\ or\ isn't\ valid\ in\ this\ context
     | unrecognized\ configuration\ parameter
-    | Coordinator\ is\ stuck
     )
     .* $
     """,
     re.VERBOSE | re.MULTILINE,
 )
 
+# Panics are multiline and our log lines of multiple services are interleaved,
+# making them complex to handle in regular expressions, thus handle them
+# separately.
+PANIC_START_RE = re.compile(rb"^(?P<service>.*) \| thread '.*' panicked at ")
+PANIC_CONTINUED_RE = re.compile(rb"^(?P<service>.*) \| (?P<msg>.*)$")
```

Comment:
Would it be possible to entirely disregard the rest of the panic after the first line? Or you want to be able to `ci-regexp` on some function name that is in the callstack?

Reply:
The callstack doesn't matter and we ignore it. The panics are now split so the file and message are on two lines: https://blog.rust-lang.org/2023/10/05/Rust-1.73.0.html#cleaner-panic-messages
```
thread 'main' panicked at src/main.rs:3:5:
oh no! "ferris.txt" not found!
```
So in order to be able to put the `panicked` and the actual panic message together, all this logic is required.

Reply:
The alternative would be for us to overwrite the default panic handler and change it back.